test/core/transport/metadata_test.cc

```diff
@@ -286,6 +289,13 @@ static void test_user_data_works(void) {
   GPR_ASSERT(grpc_mdelem_get_user_data(md, gpr_free) == ud1);
   GRPC_MDELEM_UNREF(md);
 
+  md_static = GRPC_MAKE_MDELEM(&grpc_static_mdelem_table[0],
+                               GRPC_MDELEM_STORAGE_STATIC);
+  grpc_mdelem_set_user_data(md_static, gpr_free, ud3);
+  GPR_ASSERT(grpc_mdelem_get_user_data(md_static, gpr_free) == 
```

Comment:
Why wouldn't it hold true? Aren't we setting md_static to have the user data ud3?

Reply:
for static metadata, we do not make changes to the user data associated with it

Reply:
Why do we allow users to call grpc_mdelem_set_user_data on static metadata if it's a no-op? That seems like a pretty confusing API.

Reply:
since, it is internal to core, I think it's fine. I haven't looked at if there's any behavior depending on this but we could probably change that if we feel it's necessary.

Reply:
I add this because I  take a look at the result after running the test coverage,seems it is not so necessary, I'll close this PR.